src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -443,12 +443,32 @@ public SAMFileHeader getHeader() {
      */
     private SamFileHeaderMerger createHeaderMerger() {
         List<SAMFileHeader> headers = new ArrayList<>(readers.size());
+        final Set<SAMFileHeader.SortOrder> sortOrders = new HashSet<>(SAMFileHeader.SortOrder.values().length);
         for ( Map.Entry<SamReader, CloseableIterator<SAMRecord>> readerEntry : readers.entrySet() ) {
-            headers.add(readerEntry.getKey().getFileHeader());
+            final SAMFileHeader h = readerEntry.getKey().getFileHeader();
+            headers.add(h);
+            sortOrders.add(h.getSortOrder());
         }
 
-        // TODO: don't require coordinate ordering
-        SamFileHeaderMerger headerMerger = new SamFileHeaderMerger(SAMFileHeader.SortOrder.coordinate, headers, true);
+        // unknown order is removed and assumes that it is the same as the others
+        final boolean containUnknown = sortOrders.remove(SAMFileHeader.SortOrder.unknown);
```

Comment:
I think that a refactoring for sort orders in htsjdk will be required eventually, if https://github.com/samtools/hts-specs/issues/204 is added to the specs. Anyway, I will treat for this fix the unknown as any other sort order.